examples/swe_agent/main.ipynb

```diff
@@ -0,0 +1,304 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Example with SWE-agent\n",
+    "\n",
+    "SWE-agent(SoftWare Engineering Agent) is an agent designed for solving real world software engineering problems, such as fixing github issues.\n",
+    "More details can be found in their [homepage](https://swe-agent.com/) and related [github repo](https://swe-agent.com/).\n",
+    "\n",
+    "In the example here, we partially implement the SWE-agent, and provide a simple example of how to use the implemented SWE-agent to fix a bug in a python file.\n",
+    "You should note that currently how to enable agents with stronger programming capabilities remains an open challenge, and the performance of the paritially implemented SWE-agent is not guaranteed.\n",
+    "\n",
+    "## Prerequisites\n",
+    "\n",
+    "- Follow [READMD.md](https://github.com/modelscope/agentscope) to install AgentScope. \n",
+    "- Prepare a model configuration. AgentScope supports both local deployed model services (CPU or GPU) and third-party services. More details and example model configurations please refer to our [tutorial]\n",
+    "- Understand the ServiceFactory module and how to use it to pre-process the tool functions for LLMs. You can refer to the [ReAct agent example](../conversation_with_react_agent/main.ipynb).\n",
+    "\n",
+    "## Note\n",
+    "\n",
+    "- The example is tested with the following models. For other models, you may need to adjust the prompt.\n",
+    "    - gpt-4\n",
+    "- How to enable agents with stronger programming capabilities remains an open challenge, and the current implementations are not perfect. Please feel free to explore it yourself."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 1: Initialize the AgentScope environment and SWE-agent\n",
+    "\n",
+    "Here we init the agentscope environment and load the SWE-agent.\n",
+    "\n",
+    "The code of SWE-agent is in `swe_agent.py`, and the related prompts are in `swe_agent_prompts.py`.\n",
+    "\n",
+    "If you are interested in the details, please refer to the code and the origianl SWE-agent repo [here](https://github.com/princeton-nlp/SWE-agent)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from swe_agent import SWEAgent\n",
+    "\n",
+    "import agentscope\n",
+    "from agentscope.models import load_model_by_config_name\n",
+    "agentscope.init(\n",
+    "    model_configs=\"xxx.json\", # fill you model config file path here\n",
+    ")\n",
+    "loaded_model = load_model_by_config_name('gpt-4')\n",
```

Comment:
Why we need the loaded model here?
Leave the config name to be a variable rather than "gpt-4"